tests/run-make/lib-trait-for-trait-no-ice/rmake.rs

```diff
@@ -0,0 +1,13 @@
+// Inside a library, implementing a trait for another trait
+// with a lifetime used to cause an internal compiler error (ICE).
+// This test checks that this bug does not make a resurgence -
+// first by ensuring successful compilation, then verifying that
+// the lib crate-type flag was actually followed.
+// See https://github.com/rust-lang/rust/issues/18943
+
+use run_make_support::{count_rlibs, rustc};
+
+fn main() {
+    rustc().input("foo.rs").crate_type("lib").run();
+    assert_eq!(count_rlibs("foo"), 1);
```

Comment:
Problem: I don't like relying on count here. For vast majority of tests, we should know *exactly* which output artifacts we generate and the expected exact output artifact not existing is a bug (in the test, in rustc, or both).

I would expect something more like

```rs
    fs_wrapper::remove_file(rust_lib_name("foo"));
```